src/foam/core/Listener.js

```diff
@@ -113,6 +130,15 @@ foam.CLASS({
         configurable: true,
         enumerable: false
       });
+
+      if ( on.length > 0 ) {
+        var listener = obj[this.name];
+        for ( var i = 0 ; i < on.length ; i++ ) {
+          var topic = [on[i].split('.')[1]];
+          obj.onDetach(obj.sub.apply(obj, topic.concat(listener)));
```

Comment:
Isn't obj the prototype, not the individual objects? I think this block should be in the block of code that creates the per-listener objects in lines 113-128.

Reply:
I try it, but it did not work.

Reply:
I don't see how this can work. The listener initialization needs to be on the object initialization rather than when the prototype is created. To do this, axioms can implement the initObject() method which gets called when each new object of this type is created. In this method it should hookup the subscriptions to call this listener.